mapping.Rmd

```diff
@@ -31,1098 +33,1520 @@ options(scipen = 999)
+    # For now we need to rename a few columns as urbnthemes has to be
+    # updated
+    grid = urbnthemes::urbn_geofacet %>%
+      rename(
+        code = state_code,
+        name = state_name
+      )
+  )
+```
 
-The two biggest challenges in sampling coordinates are usually: 
-1) Finding the correct shapefile because geographies change over time and geographic features like water exist in the middle of geographies. 
-2) Using and merging the shapefile in R because shapefiles are usually complicated hierarchical data structures. 
+Interactive geofacets of the United States have been used in Urban
+Features like [A Matter of
+Time](https://apps.urban.org/features/long-prison-terms/trends.html)
+which included geofaceted line charts showing trends in incarceration by
+state. Static geofacets of the United States were included in [Barriers
+to Accessing Homeownership Down Payment, Credit, and
+Affordability](https://www.urban.org/sites/default/files/publication/94801/barriers-to-homeownership-down-payments-credit-access-and-affordability_3.pdf)
+by the Housing Finance Policy Center.
+
+## Cartograms
+
+Cartograms are a modified form of a choropleth map with intentionally
+distorted sizes that map to a variable in your data. Below we create a
+cartogram with `library(cartogram)` where the state sizes are
+proportional to the population.
+
+```{r cartogram-example, cache = TRUE}
+library(cartogram)
+
+set_urbn_defaults(style = "print")
+
+chip_with_geographies_weighted <- chip_with_geographies %>%
+  # Note column name needs to be in quotes for this package
+  cartogram_cont(weight = "population")
+
+ggplot() +
+  geom_sf(
+    data = chip_with_geographies_weighted,
+    # Color in states by chip percentages
+    aes(fill = chip_pct)
+  )
+```
 
-IPUMS helps with first issue because it is a good [source](https://usa.ipums.org/usa/volii/boundaries.shtml) for high-quality shapefiles.
+## Interactive Maps
 
-The new version of RStudio helps with the second issue because it boasts a new feature that visualizes the branches of the data and helps select desired variables. Click on the shapefile in RStudio to bring up a map of the data. Mouse over the desired variable and click the little logo that appears to the far right. 
+Interactive maps can be a great exploratory tool to explore and
+understand your data. And luckily there are a lot of new R packages that
+make it really easy to create them. Interactive maps are powerful but
+**we do not recommend them for official use in Urban publications** as
+getting them in Urban styles and appropriate basemaps can be tricky
+(reach out to
+[anarayanan\@urban.org](mailto:anarayanan@urban.org){.email} if you
+really want to include them).
 
+### `library(mapview)`
 
-![](mapping/www/images/shapefile.png)
+`library(mapview)` is probably the most user friendly of the interactive
+mapping R libraries. All you have to do to create an interactive map is:
```

Comment:
Hawaii is in Mexico for all of these examples

Reply:
Sigh yes it is

Reply:
For now I'm removing bc we want the focus to be on how to create interactive maps. And I'm adding a note to directly use tigris::states() if ppl want AL/HI in the right position